games/static/games/proto.js

```diff
@@ -4,31 +4,56 @@ class Game {
     this.setup = setup;
     this.grid = document.querySelector(".grid");
     this.startButton = document.querySelector("#start");
+    this.restartButton = document.querySelector("#restart");
     this.isRunning = false
   }
   start() {
-    let alienMovementId = 0;
+    let gameId = 0;
+    let cleanId = 0;
+    let start
+    let restart
     let board = new Board(this.grid, BOARD_WIDTH, this.setup);
-    this.startButton.addEventListener("mousedown", () => {
-      if (alienMovementId) {
-        clearInterval(alienMovementId);
-        alienMovementId = null;
-        this.isRunning = false
+    this.startButton.addEventListener("mousedown", start = () => {
+      if (gameId) {
+        clearInterval(gameId);
+        gameId = null;
+        this.gamePause();
       } else {
-        this.isRunning = true
-        alienMovementId = setInterval(() => {
-          if (board.alienLocations.length === 0){
-            clearInterval(alienMovementId);
-            this.isRunning = false
-          } 
-          board.moveAliens();
-        }, 500);
+          this.gameRun();
+          gameId = setInterval(() => {
+            if (board.noAliensLeft()){
+              this.grid.innerHTML = '';
+              this.grid.classList.add('youwin');
+              clearInterval(gameId);
+              this.gamePause();
+            } 
+            board.moveAliens();
+          }, 500);
       }
     });
     board.draw(this.setup);
     this.controls(board);
+    this.restartButton.addEventListener("mousedown", restart = () => {
+      clearInterval(gameId)
+      this.restart(board ,gameId, start, restart)
+    })
+  }
+  restart(board, gameId, start, restart) {
+    this.startButton.removeEventListener("mousedown", start)
+    this.restartButton.removeEventListener("mousedown", restart)
+    clearInterval(gameId);
+    gameId = null;
+    board = null
+    this.grid.innerHTML = '';
```

Comment:
again - low level with high level code are mixed. extract this one to a function